Updates to Actor Importer

snap01 opened the pull request:
## Description.
- Optionally check for skills/weapons in compendiums
- Changes to the regular expressions to support more examples that follow similar structure
- Support for Shotgun and half DB damage values
- Work out the header and add remaining text to the occupation
- Disable auto hp, mp, mov, db, and/or build if text overwrites this
- Add any lines not imported to GM Notes

resolves #829

## Types of Changes.
- [ ] Bug fix (non-breaking change which fixes an issue).
- [X] New feature (non-breaking change which adds functionality).
- [ ] Breaking change (fix or feature that would cause existing functionality to change).

snap01 committed:
Updates to Actor Importer
- Optionally check for skills/weapons in compendiums
- Changes to the regular expressions to support more examples that follow similar structure
- Add any lines not imported to GM Notes
- Support for Shotguns and half DB
- Work out the header and add remaining text to the occupation
- Guess combat location in text
- Disable auto hp, mp, mov, db, and/or build if text uses a different one

snap01 committed:
Combat percent doesn't always show hard and extreme values

snap01 committed:
Add 'Bite' to the guess start of combat section, and allow multiple asterisks after weapon names

snap01 committed:
Fix incorrectly removing closing bracket if name is not specialization (name)

pconcepcion commented on module/apps/actor-importer-dialog.js:
```diff
@@ -93,7 +96,9 @@ export class CoC7ActorImporterDialog extends Dialog {
+  static async create (data = {}) {
+    data.languages = CoC7ActorImporterRegExp.getTranslations()
+    data.language = CoC7ActorImporterRegExp.getLanguage()
     const html = await renderTemplate(
```
getLanguage looks like needs the language as a paramter

snap01 replied:
The function has been renamed and the default parameter is explicitly set

pconcepcion commented on module/apps/actor-importer-dialog.js:
```diff
@@ -39,6 +39,9 @@ export class CoC7ActorImporterDialog extends Dialog {
     inputs.lang = $('#coc-entity-lang')
       .val()
       .trim()
```
Would be nice to validate the input it's one of the allowed language strings.

snap01 replied:
Before the language is set it runs through the ```checkLanguage ```function defaulting to "en" if not found

pconcepcion commented on module/apps/actor-importer.js:
```diff
@@ -612,29 +886,35 @@ export class CoC7ActorImporter {
    */
-  async convert7E (creature) {
+  async convert7E (npc) {
     if (CONFIG.debug.CoC7Importer) {
```
I liked more _creature_ since for me works for both _monsters_ and _npcs_

snap01 replied:
It was only renamed to keep the naming consistent in the importer, the variable name has been changed back to creature

pconcepcion commented on module/apps/actor-importer.js:
```diff
@@ -1,306 +1,528 @@
+      if (value !== Number(npc.hpMax)) {
+        updateData['data.attribs.hp.auto'] = false
+        updateData['data.attribs.hp.value'] = value
+        updateData['data.attribs.hp.max'] = value
```
this 3 `updateData` lines could be extracted to a function to simplify all this ifs

snap01 replied:
The numeric values now call a function to change updateData

snap01 committed:
Dialog checks that language exists
Fixes to zh-TW regular expressions
Change references to npc back to creature in convert7E function
Move checks to disable auto attributes to a separate function for numeric values
npm run format

snap01 committed:
Merge branch 'develop' into actor-importer-v2

snap01 committed:
Update actor-importer-dialog.js

pconcepcion commented on module/apps/actor-importer-regexp.js:
```diff
@@ -1,269 +1,491 @@
-/* global CONFIG, game */
+/* global game */
 import { CoC7Utilities } from '../utilities.js'
 
```
This is probably the most cryptic and complex file of the importer and I think it would be nice to have a general comment at the beginning explaining how it works.

snap01 replied:
A comment has been added at the top of the file explaining each keys and translations entry. The weapon types have been moved to keys.

pconcepcion commented on module/apps/actor-importer-regexp.js:
```diff
@@ -1,269 +1,491 @@
+  },
+  es: {
+    age: '(?<age>\\d+)\\s*' + 'a[ñÑ]os' + '(?![a-z])[,\\s]*',
+    occupation: '[,\\s]*' + 'Occupation' + ':?\\s+(?<occupation>.+)[,\\s\n]*',
```
This probably should be _Ocupación_ for the Spanish version.

snap01 replied:
This is a new key so was copied from the English text. A new comment has been added above changes that need made
/* NEW KEY BELOW - TRANSLATION REQUIRED */
The explanation to look for it was added to the comments at the top of the file.

pconcepcion replied:
Great! This will help for sure! I'll see if I can add some of the translations.

snap01 committed:
Add text header to actor-importer-regexp.js to explain keys and translations
Default importer to location Items / World / Modules / System

snap01 committed:
Update text in actor importer regexp comments

pconcepcion commented on module/apps/actor-importer-regexp.js:
```diff
@@ -1,269 +1,518 @@
+    newCombatHeader: '\n' + 'Combate' + '\n',
+    sectionSkills: '\n(?:' + 'Habilidades' + '(?:\\s*\\([^\\)]+\\))?)[:\n]',
+    sectionLangauges: '\n(?:' + 'Idiomas|Lenguajes|Lenguas' + ')[:\n]',
+    sectionSpells: '\n(?:' + 'Conjuros' + ')[:\n]',
```
This should be `Conjuros|Hechizos`

snap01 replied:
Updated

pconcepcion commented on module/apps/actor-importer-regexp.js:
```diff
@@ -1,269 +1,518 @@
+    sanLossNone: 'no',
+    fulldb: '(' + 'BD|bd|bonificación de daño' + ')',
+    /* NEW KEY BELOW - TRANSLATION REQUIRED */
+    halfdb: '(' + '½|half' + ')',
```
`half` could be `medio daño` or `mitad` or `daño a la mitad`.

snap01 replied:
Updated

snap01 committed:
Update Spanish actor importer keys
If the occupation includes a full stop only include up to that
